frontend/src/v5/ui/routes/viewer/groups/groupItem/groupSetItem.component.tsx

```diff
@@ -0,0 +1,112 @@
+/**
+ *  Copyright (C) 2022 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { isV5 } from '@/v4/helpers/isV5';
+import { GroupsActionsDispatchers } from '@/v5/services/actionsDispatchers/groupsActions.dispatchers';
+import { GroupsHooksSelectors } from '@/v5/services/selectorsHooks/groupsSelectors.hooks';
+import { ChevronButton } from '@controls/chevronButton';
+import { SyntheticEvent } from 'react';
+import { getGroupNamePath } from '../groupList.helpers';
+import { CollapsibleIconV4, GroupSetName, GroupsSetTreeListItemComponent } from './groupSetItem.styles';
+
+const getGroupSetData = (groupSet, overrides, highlights) => {
+	const data = groupSet.children.reduce((partialData, groupOrGroupSet:any) => {
+		// eslint-disable-next-line prefer-const
+		let { overriden, descendants, highlighted } = partialData;
+		let childGroupSetData = null;
+		let childHighlight = null;
+
+		if (groupOrGroupSet.children) {
+			childGroupSetData = getGroupSetData(groupOrGroupSet, overrides, highlights);
+			childHighlight = childGroupSetData.highlight;
+			Array.prototype.push.apply(descendants, childGroupSetData.descendants);
```

Comment:
If you `childGroupSetData.descendants` is never null, but at most an empty array, you can simply write 
`descendants.push(...childGroupSetData.descendants)`
which I think is a bit clearer